server/src/main/java/org/opensearch/tasks/CancellableTask.java

```diff
@@ -50,6 +50,14 @@ public abstract class CancellableTask extends Task {
     private volatile String reason;
     private final AtomicBoolean cancelled = new AtomicBoolean(false);
     private final TimeValue cancelAfterTimeInterval;
+    /**
+     * The time this task was cancelled as a wall clock time since epoch ({@link System#currentTimeMillis()} style).
+     */
+    private Long cancellationStartTime = null;
+    /**
+     * The time this task was cancelled as a relative time ({@link System#nanoTime()} style).
+     */
+    private Long cancellationStartTimeNanos = null;
```

Comment:
Can we use primitive long instead?

Reply:
I think it would be better to keep it Long object considering this can be nullable, also suggested by @andrross [here](https://github.com/opensearch-project/OpenSearch/pull/7445#discussion_r1199296121). Otherwise would need something like -1 to check whether this task is cancelled or not.

Reply:
We are maintaining a `cancelled ` variable anyways

Reply:
We do in certain places. But reading via streamInput or writing into one, wouldn't it make sense to use Optional Long here and mark it as null? Otherwise using primitive, we are forced to use something like -1 to initialize it.

Reply:
+1 on using the primitive. Just wondering if we should just put 0, and serialize without checking cancelled to keep things simple. 0 does explicitly indicate that the cancellation has not been triggered for this, hence the value is 0

Reply:
> 0 does explicitly indicate that the cancellation has not been triggered for this

I disagree. 0 indicates that cancellation was triggered on January 1, 1970.

Reply:
I don't have a strong opinion on whether we should use `@Nullable`, primitive `long` or `Long`.  But it it does seem like the rest of the code uses a combination of an annotation and a non-primitive type?

```
            @Nullable
            private final Integer position;
```

Definitely agree that 0 is not the same as `null`.